### DISCUSSION TITLE
how are user profile attributes retrieved?

### DISCUSSION THREAD

**Discussion by mscurtescu** [ory/hydra#2695] [Q&A] [answered]
For Id Tokens and UserInfo responses, how are relevant user attributes retrieved? For example, if the `address` scope is present, where are the relevant address claims coming from?

**Comment by vinckr** [comment, marked as answer]
Can you give me some more information on what you are trying to do ? Or point me to somewhere in the documentation?

UserInfo API doc https://www.ory.sh/hydra/docs/reference/api/#operation/userinfo

> This endpoint returns the payload of the ID Token, including the idTokenExtra values, of the provided OAuth 2.0 Access Token.

**Reply by mscurtescu** [reply]
Where are all the UserInfo claims coming from? How is Hydra fetching them?

**Reply by vinckr** [reply]
What do you mean where is the claim coming from? 
The API doc describes how to fetch it. The UserInfo endpoint mirrors the data from `session.id_token`.

Check out the [documentation](https://github.com/ory/hydra/blob/24f91ab7038e4c612db5af86ff0f3ee5eb03c878/docs/versioned_docs/version-v1.10/concepts/openid-connect-oidc.mdx#userinfo).

```
GET https://ory-hydra:4444/userinfo
Authorization: bearer access-token.xxxx

{
 "acr": "oauth2",
 "sub": "xxx@xxx.com"
}
```

By default userinfo only includes `acr` and `sub` claims, but you can add more to it:

`Any information set to the key session.id_token during accepting the consent request will also be included.`
